packages/@aws-cdk/cdk/lib/cloudformation/condition.ts

```diff
@@ -26,35 +25,52 @@ export class Condition extends Referenceable {
- * create stack resources. These conditions are evaluated based on input parameters that you
- * declare when you create or update a stack. After you define all your conditions, you can
- * associate them with resources or resource properties in the Resources and Outputs sections
- * of a template.
+ * Represents a CloudFormation element that can be used within a Condition.
  *
- * You define all conditions in the Conditions section of a template except for ``Fn::If`` conditions.
- * You can use the ``Fn::If`` condition in the metadata attribute, update policy attribute, and property
- * values in the Resources section and Outputs sections of a template.
+ * You can use intrinsic functions, such as ``Fn.conditionIf``,
+ * ``Fn.conditionEquals``, and ``Fn.conditionNot``, to conditionally create
+ * stack resources. These conditions are evaluated based on input parameters
+ * that you declare when you create or update a stack. After you define all your
+ * conditions, you can associate them with resources or resource properties in
+ * the Resources and Outputs sections of a template.
  *
- * You might use conditions when you want to reuse a template that can create resources in different
- * contexts, such as a test environment versus a production environment. In your template, you can
- * add an EnvironmentType input parameter, which accepts either prod or test as inputs. For the
- * production environment, you might include Amazon EC2 instances with certain capabilities;
- * however, for the test environment, you want to use less capabilities to save costs. With
- * conditions, you can define which resources are created and how they're configured for each
- * environment type.
+ * You define all conditions in the Conditions section of a template except for
+ * ``Fn.conditionIf`` conditions. You can use the ``Fn.conditionIf`` condition
+ * in the metadata attribute, update policy attribute, and property values in
+ * the Resources section and Outputs sections of a template.
+ *
+ * You might use conditions when you want to reuse a template that can create
+ * resources in different contexts, such as a test environment versus a
+ * production environment. In your template, you can add an EnvironmentType
+ * input parameter, which accepts either prod or test as inputs. For the
+ * production environment, you might include Amazon EC2 instances with certain
+ * capabilities; however, for the test environment, you want to use less
+ * capabilities to save costs. With conditions, you can define which resources
+ * are created and how they're configured for each environment type.
  */
-export class FnCondition extends CloudFormationToken {
-  constructor(type: string, value: any) {
-    super({ [type]: value });
-  }
-}
+export interface IConditionExpression {
+  /**
+   * Returns a JSON node that represents this condition expression
+   */
+  resolve(): any;
```

Comment:
Why `resolve()` and not a more specific fucntion name? If you wanted `resolve()` you could have accepted a Token.

Reply:
But I wanted the Condition construct to implement it, so I had to define it as an interface. And I want the recursive resolution behavior of tokens to apply to full expressions, so I just faked it.
We should probably define an `IToken` with a resolve method so (reopened #79)